dependencydiff/raw_dependencies.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencydiff
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/ossf/scorecard/v4/checks"
+	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/clients/githubrepo"
+	"github.com/ossf/scorecard/v4/clients/githubrepo/roundtripper"
+	"github.com/ossf/scorecard/v4/pkg"
+)
+
+// Dependency is a raw dependency fetched from the GitHub Dependency Review API.
+type dependency struct {
+	// Package URL is a short link for a package.
+	PackageURL *string `json:"package_url"`
+
+	// SourceRepository is the source repository URL of the dependency.
+	SourceRepository *string `json:"source_repository_url"`
+
+	// ChangeType indicates whether the dependency is added, updated, or removed.
+	ChangeType *pkg.ChangeType `json:"change_type"`
+
+	// ManifestPath is the path of the manifest file of the dependency, such as go.mod for Go.
+	ManifestPath *string `json:"manifest"`
+
+	// Ecosystem is the name of the package management system, such as NPM, GO, PYPI.
+	Ecosystem *string `json:"ecosystem"`
+
+	// Version is the package version of the dependency.
+	Version *string `json:"version"`
+
+	// Name is the name of the dependency.
+	Name string `json:"name"`
+}
```

Comment:
You shouldn't need a new struct for parsing the response JSON. `pkg.DependencyCheckResult` can be used here directly.

Reply:
@laurentsimon and I discussed this before. The `dependency` struct here with JSON tags in snakecase (underslines) is used for receiving raw data from the GH API, while the `pkg.DependencyCheckReult` is the exported one with JSON tags in camelcase following the Scorecard repo's naming convention.